treeherder/config/settings.py

```diff
@@ -27,6 +27,9 @@ def server_supports_tls(url):
 
 # Default to retaining data for ~4 months.
 DATA_CYCLE_DAYS = env.int("DATA_CYCLE_DAYS", default=120)
+# Determines the number of jobs we try to delete per iteration when
+# cycling data
+DATA_CYCLE_CHUNK_SIZE = env.int("DATA_CYCLE_CHUNK_SIZE", default=1000)
```

Comment:
So in my testing we're still OOM'ing with this value: I guess going across all those foreign key relations can be expensive. Reducing it to 100 seems to work (it's steadily expiring data on inbound, at least).